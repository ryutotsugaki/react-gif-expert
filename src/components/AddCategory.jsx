import { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = (props) =>{
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (evento) =>{
        setInputValue(evento.target.value);
        // console.log(evento.target.value);
    };

    const onSubmit = (evento) =>{
        evento.preventDefault();
        if(inputValue.trim().length <= 1) return;
        // console.log(inputValue);
        props.onNewCategory(inputValue.trim());
        setInputValue('');
    };
    return (
        <form onSubmit={ onSubmit }>
            <input
            
            value= { inputValue }
            onChange={ onInputChange }
            type="text"
            placeholder="Buscar gifs guays"

             />
        </form>
    );

};


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}