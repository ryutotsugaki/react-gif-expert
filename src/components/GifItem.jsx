export const GifItem = ({title, url, id})=>{
    console.log(title);
    return (
        <div className="card">
            <img src={url} alt={title} title={ title } />
            <p> {title} </p>
        </div>
    );
};