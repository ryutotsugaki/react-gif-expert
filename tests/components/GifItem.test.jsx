import { render, screen } from '@testing-library/react'
import { GifItem } from "../../src/components/GifItem";


describe('Las pruebas en el componente <GifItem />', () => {
    const title = "OverWatch"
    const url = "https://www.google.com/"
    // Prueba snapshot con GifItem
    test('Hacer un snapshot con el GifItem debe hacer el match con el snapshot', () => {
        const { container } = render (<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    });

    // prueba de debe mostrar la Imagen con el URL y el ALT indicado

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifItem title={ title } url={ url } /> );

        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
        
    });

    // Prueba de debe de mostrar el titulo en el componente

    test('debe de mostrar el titulo en el componente', () => {
        render( <GifItem title={ title } url={ url } /> );
        expect( screen.getByText(title) ).toBeTruthy();
    });

});