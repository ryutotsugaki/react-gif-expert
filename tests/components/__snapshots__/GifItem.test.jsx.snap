// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Las pruebas en el componente <GifItem /> Hacer un snapshot con el GifItem debe hacer el match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="OverWatch"
      src="https://www.google.com/"
      title="OverWatch"
    />
    <p>
       
      OverWatch
       
    </p>
  </div>
</div>
`;
